base/src/schema.test.ts

```diff
@@ -915,6 +998,229 @@ describe('schema', () => {
 
       expect(() => Schema.validateSchema(schema)).not.toThrow()
     })
+
+    it('throws a validation error when type is unknown', () => {
+      const schema = {
+        stuffs: { type: '' },
+      }
+
+      expect(() =>
+        Schema.validateSchema(schema as unknown as SchemaDefinitions)
+      ).toThrow('`type` is invalid or undefined')
+    })
+
+    describe('object type', () => {
+      it('throws an error when schema definitions is not an object', () => {
+        const schema = [{ stuffs: { type: 'array' } }]
+
+        expect(() =>
+          Schema.validateSchema(schema as unknown as SchemaDefinitions)
+        ).toThrow('schema has to be an object')
+      })
+
+      it('throws a validation error when default value of an object is incorrect', () => {
+        const schema = {
+          stuff: {
+            defaultValue: true,
+            type: 'object',
+          },
+        }
+
+        expect(() =>
+          Schema.validateSchema(schema as unknown as SchemaDefinitions)
+        ).toThrow('`schema` is required when type is `object`')
+      })
+
+      it('throws a validation error when object schema defaultValue doesnt equal an object', () => {
+        const schema = {
+          stuff: {
+            defaultValue: 'man',
+            schema: {},
+            type: 'object',
+          },
+        }
+
+        expect(() =>
+          Schema.validateSchema(schema as unknown as SchemaDefinitions)
+        ).toThrow('`defaultValue` should be an object')
+      })
+
+      it('throws a validation error when object schema defaultValue equals an array', () => {
+        const schema = {
+          stuff: {
+            defaultValue: ['man'],
+            schema: {},
+            type: 'object',
+          },
+        }
+
+        expect(() =>
+          Schema.validateSchema(schema as unknown as SchemaDefinitions)
+        ).toThrow('`defaultValue` should be an object')
+      })
+
+      it('validates an object schema', () => {
```

Comment:
```suggestion
      it('returns schema when valid', () => {
```

And update the test to check for the returned schema